### PR TITLE
Standard library task

### PR DESCRIPTION
**Pull request by AnnGuden**

### CODE REVIEW

File: 2-standard-library/activityMonitor.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+const logFilePath = 'activityMonitor.log';
+
+let logData = '';
+
+const getCurrentTime = () => {
+    return Math.floor(Date.now() / 1000);
+}
+
+const getProcessCommand = () => {
+    if (process.platform === 'win32') {
```
Comment (andrei-savelev-epam) [inline 2-standard-library/activityMonitor.js]: Usually, it's better to keep such string values in the constants. As well as the return statements below.

File: 2-standard-library/activityMonitor.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        return 'powershell "Get-Process | Sort-Object CPU -Descending | Select-Object -Property Name, CPU, WorkingSet -First 1 | ForEach-Object { $_.Name + \' \' + $_.CPU + \' \' + $_.WorkingSet }"';
+    } else {
```
Comment (andrei-savelev-epam) [inline 2-standard-library/activityMonitor.js]: You can simplify this function by avoiding the esle statment.

File: 2-standard-library/activityMonitor.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+const childProcess = require('child_process');
```
Comment (andrei-savelev-epam) [inline 2-standard-library/activityMonitor.js]: [minor]: One of the bestpractice is using the "node" prefix pattern to indicate the node related libraries, like this
`const childProcess = require('node:child_process');`

File: 2-standard-library/activityMonitor.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        let outputTrimmed = stdout.trim();
+        let processInfo = truncateText(outputTrimmed, process.stdout.columns - 1);
+        process.stdout.clearLine();
+        process.stdout.write(processInfo + '\r');
+
+        let time = getCurrentTime();
+        logData += `${time} : ${outputTrimmed}\n`;
+
+        if (error !== null) {
+            console.log(`error: ${error}`);
```
Comment (andrei-savelev-epam) [inline 2-standard-library/activityMonitor.js]: Currently, if childProcess.exec() fails, we log the error message to the console, but the error itself isn't caught or handled in any way--it simply logs the error and then continues with the rest of the script.

My suggestion is to throw an error when childProcess.exec() fails and then handle that exception in your main function.

File: 2-standard-library/activityMonitor.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+const truncateText = (text, maxlength) => {
+    if (text.length > maxlength) {
+        return text.slice(0, maxlength - 3) + '...';
+    }
+    return text;
+}
+
+const monitorSystem = () => {
```
Comment (andrei-savelev-epam) [inline 2-standard-library/activityMonitor.js]: You can simplify this function by the following:
1. use the async/awit pattern here
2. use of promisify() from the util module of node.js to create a promise-based version of exec().
3. await the outcome of exec(command).
4. replace the callback logic with a try-catch block to handle errors. I have a commented it as well

File: 2-standard-library/activityMonitor.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        let processInfo = truncateText(outputTrimmed, process.stdout.columns - 1);
+        process.stdout.clearLine();
+        process.stdout.write(processInfo + '\r');
+
+        let time = getCurrentTime();
+        logData += `${time} : ${outputTrimmed}\n`;
+
+        if (error !== null) {
+            console.log(`error: ${error}`);
+        }
```
Comment (andrei-savelev-epam) [inline 2-standard-library/activityMonitor.js:56]: It could be better if errors are also logged in the file

**Review by andrei-savelev-epam — COMMENTED** [review]
Left a few comments there.

File: 2-standard-library/activityMonitor.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    });
+}
+
+// Monitor with a refresh rate of 10 times per second
+setInterval(monitorSystem, 100);
+
+setInterval(() => {
+    logSystemActivity(logData);
+    logData = '';
```
Comment (andrei-savelev-epam) [inline 2-standard-library/activityMonitor.js]: Please, keep in mind that fs.appendFile is an async. So, I would rather suggest you to one of the solutions:
1. use promises as well and clear the log data on success/catch
2. keep file opened when the program is started and just write data on the fly, and then close the file on exit.

File: 2-standard-library/activityMonitor.js (hunk 1 above)
Comment (AnnGuden) [reply 2-standard-library/activityMonitor.js]: Done

File: 2-standard-library/activityMonitor.js (hunk 2 above)
Comment (AnnGuden) [reply 2-standard-library/activityMonitor.js]: Done

File: 2-standard-library/activityMonitor.js (hunk 3 above)
Comment (AnnGuden) [reply 2-standard-library/activityMonitor.js]: Done

File: 2-standard-library/activityMonitor.js (hunk 5 above)
Comment (AnnGuden) [reply 2-standard-library/activityMonitor.js]: Done

File: 2-standard-library/activityMonitor.js (hunk 6 above)
Comment (AnnGuden) [reply 2-standard-library/activityMonitor.js:56]: Done

File: 2-standard-library/activityMonitor.js (hunk 7 above)
Comment (AnnGuden) [reply 2-standard-library/activityMonitor.js]: Done

**Review by andrei-savelev-epam — APPROVED** [review]

**Pull request merged**